ragna/deploy/_ui/left_sidebar.py

```diff
@@ -59,6 +59,10 @@ def refresh(self):
 
     @pn.depends("refresh_counter", "chats", "current_chat_id", on_init=True)
     def __panel__(self):
+        self.chats.sort(
+            key=lambda chat: chat["messages"][-1]["timestamp"], reverse=True
+        )
```

Comment:
Just need to be careful with unprepared chats that don't have any messages yet. Something like this?
```suggestion
        from datetime import datetime
        epoch = datetime(1970, 1, 1)
        self.chats.sort(
            key=lambda chat: (
                epoch if not chat["messages"] else chat["messages"][-1]["timestamp"]
            ),
            reverse=True
        )
```

Reply:
unless unprepared chats should go at the top?
```suggestion
        from datetime import datetime
        self.chats.sort(
            key=lambda chat: (
                datetime.now() if not chat["messages"] else chat["messages"][-1]["timestamp"]
            ),
            reverse=True
        )
```

Reply:
Thanks for the quick response & the suggestion really does makes sense so happy to commit it.
However i have one small doubt before that in my case, Every time I create a new chat it already has one message i.e. *How can I help you with the documents?* so the situation of chat with no messages didn't actually arise. 
Is there a way that I can create a chat with no message i.e. unprepared chat (I hope I'm referring it right)

Reply:
Having a chat with no messages, i.e. an unprepared one, can only arise if the chat is created through the API. For that, this is a two-step process, i.e. the user has to send two HTTP requests, e.g.

https://github.com/Quansight/ragna/blob/55f7fc5d3d13f321eebd16d3b0901ab35ac36bc7/scripts/add_chats.py#L51-L66

Of course this is the same in the UI, but there we can guarantee that they are performed back to back, without ever leaving the user in an intermediate state

https://github.com/Quansight/ragna/blob/55f7fc5d3d13f321eebd16d3b0901ab35ac36bc7/ragna/deploy/_ui/api_wrapper.py#L83-L98

Meaning, you'll never be able to create the issue @blakerosenthal mentioned with just the UI.

However, we also don't have a mechanism to deal with such chats in the UI yet. Meaning, if we just create a chat with the API and start the UI afterwards, you will be seeing an empty chat that you cannot prepare, but that will error when you send any prompt, because it is not prepared.

So I guess it makes sense to just move these chats at the end of the list. Ultimately, we should add functionality that triggers the preparation if at any point a user selects an unprepared chat. @arjxn-py Do you want to open an issue for that?

Reply:
> Do you want to open an issue for that?

Yes, I just opened #448, please correct me if needed.

> So I guess it makes sense to just move these chats at the end of the list.

Happy to merge @blakerosenthal's first suggestion to handle unprepared chats this way, until they are clicked on `ui` & prepared automatically.